test/integration/kubernetes_deploy_test.rb

```diff
@@ -1053,4 +1053,44 @@ def test_raise_on_yaml_missing_kind
       "  datapoint: value1"
     ], in_order: true)
   end
+
+  def test_hpa_can_be_successful
+    skip if KUBE_SERVER_VERSION < Gem::Version.new('1.8.0')
+    deploy_metric_server
+    assert_deploy_success(deploy_fixtures("hpa/hpa"))
+    assert_logs_match_all([
+      "Deploying resources:",
+      "HorizontalPodAutoscaler/hello-hpa (timeout: 300s)",
+      %r{HorizontalPodAutoscaler/hello-hpa\s+Succeeded}
+    ])
+  end
+
+  def test_hpa_can_fail_when_no_matching_deployment
+    skip if KUBE_SERVER_VERSION < Gem::Version.new('1.8.0')
+    deploy_metric_server
+    assert_deploy_failure(deploy_fixtures("hpa/hpa", subset: ["hpa.yml"]))
+    assert_logs_match_all([
+      "Deploying HorizontalPodAutoscaler/hello-hpa (timeout: 300s)",
+      "HorizontalPodAutoscaler/hello-hpa: FAILED",
+      "Final status: FailedGetScale"
+    ])
+  end
+
+  def test_hpa_can_be_pruned
+    skip if KUBE_SERVER_VERSION < Gem::Version.new('1.8.0')
+    deploy_metric_server
+    assert_deploy_success(deploy_fixtures("hpa/hpa"))
+    assert_deploy_success(deploy_fixtures("hpa/hpa", subset: ["deployment.yml"]))
+    assert_logs_match_all(['The following resources were pruned: horizontalpodautoscaler "hello-hpa"'])
+  end
+
+  private
+
+  def deploy_metric_server
```

Comment:
Is there a helper file we've been putting methods like this?

Reply:
Sort of, yes. This is being deployed globally, so I think it should happen during the initial test suite setup, which is taken care of inside `test_helper` itself. I'd add it to this module:

https://github.com/Shopify/kubernetes-deploy/blob/7b47ed81f040922ad877fc7e3035f6b82a71ea87/test/test_helper.rb#L243-L254

And then call it with the PV setup, i.e. when the file is loaded

https://github.com/Shopify/kubernetes-deploy/blob/7b47ed81f040922ad877fc7e3035f6b82a71ea87/test/test_helper.rb#L289